### PR TITLE
TST: Replace pytest.xfail

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Part of #38902.

### CODE REVIEW

File: pandas/tests/indexes/test_common.py [hunk 1]
```diff
@@ -388,7 +388,7 @@ def test_asi8_deprecation(self, index):
 @pytest.mark.parametrize("na_position", [None, "middle"])
 def test_sort_values_invalid_na_position(index_with_missing, na_position):
     if isinstance(index_with_missing, (CategoricalIndex, MultiIndex)):
-        pytest.xfail("missing value sorting order not defined for index type")
+        pytest.skip("missing value sorting order not defined for index type")
```
Comment (jreback) [inline pandas/tests/indexes/test_common.py:391]: hmm i would leave these as xfail and add_marker here. e.g. these are a missing api that we should eventually fix.

cc @jbrockmendel

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/test_common.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/indexes/test_common.py:391]: For MultiIndex we get NotImplementedError; this should be certainly be left as xfail. However for CategoricalIndex the computation is successful and sometimes the order is "correct" and sometimes not. I think this should skip instead.

Comment (rhshadrach) [reply pandas/tests/indexes/test_common.py:391]: Sorry, my remark is about `test_sort_values_with_missing`, the test immediately below this one.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**